sufia-models/app/models/concerns/sufia/collection_behavior.rb

```diff
@@ -15,10 +15,39 @@ def update_permissions
       self.visibility = "open"
     end
 
-    # Compute the sum of each file in the collection
+    # Compute the sum of each file in the collection using Solr to
+    # avoid having to proces GBs of data for a simple number. Can be
+    # cleaned up once it is confirmed to work
+    #
     # Return an integer of the result
     def bytes
-      members.reduce(0) { |sum, gf| sum + gf.content.size.to_i }
-    end
+      query = "has_model_ssim:#{file_model}"
+      # Cribbed from Hydra::Collections to speed up the resolution
+      # process
+      args = {
+        fq: "{!join from=hasCollectionMember_ssim to=id}id:#{self.id}",
+        fl: "id, #{file_size_field}",
+        rows: members.count
+      }
+
+      files = ActiveFedora::SolrService.query(query, args)
+      files.reduce(0) { |sum, f| sum += f[file_size_field].to_i }
+   end
+
+   # Field to look up when locating the size of each file in Solr.
+   # Override for your own installation if using something different
+   def file_size_field
```

Comment:
That's a good question - is there ever any other situation where another class might need access to it? If not then it makes logical sense to convert it to a protected method.